src/main/java/com/googleinterns/zoomtube/servlets/TranscriptLanguageServlet.java

```diff
@@ -0,0 +1,144 @@
+  private static final String API_URL = "http://video.google.com/timedtext";
+  private static final String API_PARAM_TYPE = "type";
+  private static final String API_TYPE_LIST = "list";
+  private static final String API_PARAM_VIDEO = "v";
+
+  /* Error messages for missing or invalid video link. */
+  private static final String ERROR_MISSING_LINK = "Missing link parameter.";
+  private static final String ERROR_INVALID_LINK = "Invalid video link.";
+
+  /* Attributes for parsing the XML. */
+  private static final String TAG_TRACK = "track";
+  private static final String ATTR_LANG_CODE = "lang_code";
+  private static final String ATTR_NAME = "name";
+  private static final String ATTR_LANG_TRANSLATED = "lang_translated";
+
+  /* Name of input field used for lecture video link in lecture selection page. */
+  @VisibleForTesting static final String PARAM_LINK = "link-input";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Optional<String> error = validateGetRequest(request);
+    if (error.isPresent()) {
+      // TODO: Display a message in the client.
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, error.get());
+      return;
+    }
+
+    String videoUrl = request.getParameter(PARAM_LINK);
+    Optional<String> videoId = LectureServlet.getVideoId(videoUrl);
+    if (!videoId.isPresent()) {
+      // TODO: Display a message in the client.
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, ERROR_INVALID_LINK);
+      return;
+    }
+
+    URL transcriptLanguagesUrl = getTranscriptLanguagesUrlForVideo(videoId.get());
+    Document transcriptLanguagesDocument =
+        TranscriptParser.fetchUrlAsXmlDocument(transcriptLanguagesUrl);
+    List<TranscriptLanguage> transcriptLanguages =
+        parseTranscriptLanguages(transcriptLanguagesDocument);
+    writeTranscriptLanguages(response, transcriptLanguages);
+  }
+
+  /**
+   * Returns {@code Optional.empty()} if the video link is present. Else,
+   * returns an Optional containing an error message if it is missing.
+   */
+  private Optional<String> validateGetRequest(HttpServletRequest request) {
+    if (request.getParameter(PARAM_LINK) == null) {
+      return Optional.of(ERROR_MISSING_LINK);
+    }
+    return Optional.empty();
+  }
+
+  /**
+   * Returns the API Url used to fetch the list of transcript language options
+   * for {@code videoId}.
+   */
+  private URL getTranscriptLanguagesUrlForVideo(String videoId) throws IOException {
+    try {
+      URIBuilder urlBuilder = new URIBuilder(API_URL);
+      urlBuilder.addParameter(API_PARAM_TYPE, API_TYPE_LIST);
+      urlBuilder.addParameter(API_PARAM_VIDEO, videoId);
+      return urlBuilder.build().toURL();
+    } catch (URISyntaxException | MalformedURLException e) {
+      throw new IOException(e.getCause());
+    }
+  }
+
+  /**
+   * Returns the API Url used to fetch the list of transcript language options
```

Comment:
this JavaDoc seems to be a copy of the previous method.

Reply:
ah thanks for catching that, updated.